src/main/java/ch/njol/skript/events/EvtClick.java

```diff
@@ -97,22 +80,22 @@ public class EvtClick extends SkriptEvent {
 	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult) {
 		click = parseResult.mark == 0 ? ANY : parseResult.mark;
 		type = args[matchedPattern];
-		if (type != null && !ItemType.class.isAssignableFrom(type.getReturnType())) {
+		if (type != null && !ItemType.class.isAssignableFrom(type.getReturnType()) && !BlockData.class.isAssignableFrom(type.getReturnType())) {
 			Literal<EntityData<?>> entitydata = (Literal<EntityData<?>>) type;
 			if (click == LEFT) {
 				if (Vehicle.class.isAssignableFrom(entitydata.getSingle().getType())) {
-					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'vehicle damage' event.");
+					Skript.error("A leftclick on a vehicle entity is an attack and thus not covered by the 'click' event, use the 'vehicle damage' event.");
 				} else {
-					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'damage' event.");
+					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, use the 'damage' event.");
 				}
 				return false;
 			} else if (click == ANY) {
 				if (Vehicle.class.isAssignableFrom(entitydata.getSingle().getType())) {
-					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'vehicle damage' event. " +
-							"Change this event to a rightclick to fix this warning message.");
+					Skript.error("A leftclick on a vehicle entity is an attack and thus not covered by the 'click' event, use the 'vehicle damage' event for left clicks " +
+							"or change this event to a rightclick.");
 				} else {
-					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, but the 'damage' event. " +
-							"Change this event to a rightclick to fix this warning message.");
+					Skript.error("A leftclick on an entity is an attack and thus not covered by the 'click' event, use the 'damage' event for left clicks " +
+							"or change this event to a rightclick.");
 				}
```

Comment:
These were fine; I don't think they need changes beyond the addition of `a vehicle` like the change above.